server/services/offer.service.js

```diff
@@ -270,6 +271,25 @@ export const getOffer = async (offerId, user) => {
   return offer;
 };
 
+const generatePaymentDates = (offer) => {
```

Comment:
```suggestion
const generatePaymentSchedules = (offer) => {
```

This PR is already getting messy, lets conclude this PR by writing services tests to confirm that this function works as expected. Lets test it with various cases like:

1. When totalAmountPayable is equal to  initial payment
2. When totalAmountPayable is slightly bigger than initial payment (i.e different is not up to periodic payment) 
3. When totalAmountPayable is greater than initial Payment so that we can have a couple of payments to be made
4. Testing same values with other paymentFrequency to be sure they work as expected
5. Other tests you feel is necessary to ensure this works as expected.